freqtrade/freqtradebot.py

```diff
@@ -443,33 +442,29 @@ def process_maybe_execute_buy(self) -> None:
         except DependencyException as exception:
             logger.warning('Unable to create trade: %s', exception)
 
-    def process_maybe_execute_sell(self, trade: Trade) -> bool:
+    def process_maybe_execute_sells(self, trades: List[Any]) -> None:
         """
-        Tries to execute a sell trade
-        :return: True if executed
+        Tries to execute sell trades in a safe way
         """
-        try:
-            self.update_trade_state(trade)
-
-            if self.strategy.order_types.get('stoploss_on_exchange') and trade.is_open:
-                result = self.handle_stoploss_on_exchange(trade)
-                if result:
-                    self.wallets.update()
-                    return result
+        result = False
+        for trade in trades:
+            try:
+                self.update_trade_state(trade)
 
-            if trade.is_open and trade.open_order_id is None:
+                if (self.strategy.order_types.get('stoploss_on_exchange') and
+                        self.handle_stoploss_on_exchange(trade)):
```

Comment:
While it's not really the intend of this PR - i would want to see a test which has 2 (or more) trades, `stoploss_on_exchange=True` and `self.handle_stoploss_on_exchange` mocked in a way to return True for one trade, and False for the other (this one should have `handle_trade()` mocked to True).

It's about the logic in this function, which should call both methods once in the above sample, and call wallets.update only one time (not twice as it would have done before).

I was surprised this morning when we didn't have a test which tries to sell more than one trade at a time - and i think it makes sense to add this as part of this refactoring.

If you have difficulties with that test (i know, pytest is not you're expertise) let me know and i'll have a look ...

Reply:
> If you have difficulties with that test (i know, pytest is not you're expertise) let me know and i'll have a look ...

Yeah, that was not covered by the tests...

Yes, please, add the test -- this one still seems to be too difficult to me, all those mocking...

Reply:
it's mostly copy and paste from other tests, really ... 
and a bit of thinking about "how deep do i want the execution to go ..."

Mocking itself in this case was not difficult